File: Problem Set #2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
        {
            letters++;
        }
        else if (c == ' ')
        {
            words++;
        }
        else if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }
    words++;
    //printf("letters: %i\n", letters);
    //printf("words: %i\n", words);
    //printf("sentences: %i\n", sentences);
    float L = ((float) letters * 100) / (float) words;
    
    //printf("l: %f\n", L);

    float S = ((float) sentences * 100) / (float) words;

    //printf("s: %f\n", S);

    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    } 
    else 
    {
        printf("Grade %i\n", (int)index);
    }

}

File: Problem Set #6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 0
sentences = 0

for i in range(len(text)):
    c = text[i]
    if ((c >= 'A' and c <= 'Z') or (c >= 'a' and c <= 'z')):
        letters += 1
    elif (c == ' '):
        words += 1
    elif (c == '.' or c == '!' or c == '?'):
        sentences += 1
words += 1

L = (float(letters) * 100) / float(words)
S = (float(sentences) * 100) / float(words)

index = 0.0588 * L - 0.296 * S - 15.8
index = round(index)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {int(index)}")